app/commands/create_election.rb

```diff
@@ -36,6 +36,8 @@ def create_election
       authority: form.authority
     }
     @election = Election.create(election_attributes)
+  rescue ActiveRecord::RecordInvalid, ActiveRecord::RecordNotUnique
```

Comment:
I'm not sure, but I think that this would prevent the transaction to be aborted. I'd move this part to the `call` and allow the methods to fail, rollback the transaction and then handle the exception and broadcast the error message.

Reply:
Done! :)